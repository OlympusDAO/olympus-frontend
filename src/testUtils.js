import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
