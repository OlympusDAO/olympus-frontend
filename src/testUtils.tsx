import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
