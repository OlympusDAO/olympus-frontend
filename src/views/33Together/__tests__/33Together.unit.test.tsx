import { render } from "../../../testUtils";
import PoolTogether from "../33together";

describe("<PoolTogether/>", () => {
  it("should render component", () => {
    const { container } = render(<PoolTogether />);
    expect(container).toMatchSnapshot();
  });
});
