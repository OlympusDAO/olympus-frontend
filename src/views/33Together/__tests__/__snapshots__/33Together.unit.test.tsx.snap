// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoolTogether/> should render component 1`] = `
<div>
  <div
    id="pool-together-view"
  >
    <div
      class="MuiBox-root MuiBox-root-1 pool-prize-card"
    >
      <div
        class="MuiBox-root MuiBox-root-2 vegas-container"
      >
        <div
          class="MuiBox-root MuiBox-root-3 vegas"
        />
      </div>
      <div
        class="MuiPaper-root ohm-card MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3"
            >
              Pool Award Balance is currently 0.
            </h3>
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              Award Balance will grow at 1st rebase.
            </h4>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
          <div
            class="MuiBox-root MuiBox-root-6 pool-timer"
          >
            <div
              class="MuiBox-root MuiBox-root-39 pool-timer-unit"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 20px;"
                />
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                day
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-40 pool-timer-unit"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 20px;"
                />
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                hrs
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-41 pool-timer-unit"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 20px;"
                />
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                min
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-42 pool-timer-unit"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 20px;"
                />
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                sec
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-root-7 makeStyles-root-8  MuiPaper-elevation0 MuiPaper-rounded"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root card-header MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <div
              class="MuiBox-root MuiBox-root-10"
            >
              <h5
                class="MuiTypography-root header-text MuiTypography-h5"
              >
                3, 3 Together
              </h5>
            </div>
            <div
              class="top-right"
            >
              <div
                class="MuiBox-root MuiBox-root-11"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root info-icon MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  style="margin: 0px 5px; font-size: 16px;"
                  viewBox="0 0 20 20"
                >
                  <path
                    d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiTabs-root makeStyles-root-13 pt-tabs"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                aria-label="pool tabs"
                class="MuiTabs-flexContainer MuiTabs-centered"
                role="tablist"
              >
                <button
                  aria-controls="simple-tabpanel-0"
                  aria-label="pool-withdraw-button"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                  id="simple-tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Withdraw
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-14 PrivateTabIndicator-colorPrimary-15 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-0"
            id="simple-tabpanel-0"
            role="tabpanel"
            style="overflow: hidden;"
          >
            <div
              class="MuiBox-root MuiBox-root-18"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                style="margin: 0.5rem;"
              >
                The pool has been temporarily disabled for V2 Migration. Please withdraw your 33T
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
                style="margin: 0.5rem;"
              >
                (The 6-day early exit fee has been waived. Network fees apply.)
              </p>
              <div
                class="MuiBox-root MuiBox-root-19 pool-deposit-ui"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-20 makeStyles-root-21 undefined MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
                  style="font-size: 1.2857rem;"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Connect Wallet
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-root-7 makeStyles-root-22  MuiPaper-elevation0 MuiPaper-rounded"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root card-header MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-23"
          >
            <div
              class="MuiBox-root MuiBox-root-24"
            >
              <h5
                class="MuiTypography-root header-text MuiTypography-h5"
              >
                Prize Pool Info
              </h5>
            </div>
            <div
              class="top-right"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          
          <div
            class="MuiBox-root MuiBox-root-25 pool-data"
          >
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-50"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Winners / prize period
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 80px;"
                />
              </p>
            </div>
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-51"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Total Deposits
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 80px;"
                />
              </p>
            </div>
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-52"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Total Sponsorship
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 80px;"
                />
              </p>
            </div>
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-53"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Yield Source
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                sOHM
              </p>
            </div>
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-54"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Pool owner
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                OlympusDAO
              </p>
            </div>
            <hr
              class="MuiDivider-root"
              color="secondary"
            />
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-55"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Early Exit Fee
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 80px;"
                />
              </p>
            </div>
            <div
              class="makeStyles-root-26 data-row"
            >
              <div
                class="MuiBox-root MuiBox-root-56"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Exit Fee Decay Time
                </p>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                style="text-align: right;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style="width: 80px;"
                />
              </p>
            </div>
          </div>
          <hr
            class="MuiDivider-root"
            color="secondary"
          />
          <div
            class="MuiBox-root MuiBox-root-35"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Something not right, fren? Check Pool Together's UI below.
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-36"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-20 makeStyles-root-37 undefined MuiButton-outlinedSecondary MuiButton-disableElevation"
              href="https://app.pooltogether.com/pools/mainnet/sOHM"
              tabindex="0"
              target="_blank"
            >
              <span
                class="MuiButton-label"
              >
                sOHM Prize Pool
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M4.297 17.445h9.539c1.523 0 2.305-.78 2.305-2.28v-9.58c0-1.507-.782-2.288-2.305-2.288h-9.54c-1.523 0-2.304.773-2.304 2.289v9.578c0 1.508.781 2.281 2.305 2.281Zm.016-.968c-.875 0-1.352-.461-1.352-1.368V5.633c0-.899.477-1.367 1.352-1.367h9.5c.867 0 1.359.468 1.359 1.367v9.476c0 .907-.492 1.368-1.36 1.368h-9.5Zm7.296-4.235c.266 0 .438-.195.438-.476V7.867c0-.344-.188-.492-.492-.492H7.64c-.29 0-.47.172-.47.438 0 .265.188.445.477.445H9.53l1.133-.078-1.055.992-3.382 3.383a.476.476 0 0 0-.149.328c0 .273.18.453.453.453a.47.47 0 0 0 .344-.149L10.25 9.82l.984-1.047-.078 1.282v1.718c0 .29.18.47.453.47Z"
                    />
                  </svg>
                </span>
              </span>
            </a>
            <div
              class="marginedBtn"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-20 makeStyles-root-38 undefined MuiButton-outlinedSecondary MuiButton-disableElevation"
                href="https://community.pooltogether.com/pools/mainnet/0xEaB695A8F5a44f583003A8bC97d677880D528248/manage#stats"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiButton-label"
                >
                  sOHM Pool Details
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                      focusable="false"
                      viewBox="0 0 20 20"
                    >
                      <path
                        d="M4.297 17.445h9.539c1.523 0 2.305-.78 2.305-2.28v-9.58c0-1.507-.782-2.288-2.305-2.288h-9.54c-1.523 0-2.304.773-2.304 2.289v9.578c0 1.508.781 2.281 2.305 2.281Zm.016-.968c-.875 0-1.352-.461-1.352-1.368V5.633c0-.899.477-1.367 1.352-1.367h9.5c.867 0 1.359.468 1.359 1.367v9.476c0 .907-.492 1.368-1.36 1.368h-9.5Zm7.296-4.235c.266 0 .438-.195.438-.476V7.867c0-.344-.188-.492-.492-.492H7.64c-.29 0-.47.172-.47.438 0 .265.188.445.477.445H9.53l1.133-.078-1.055.992-3.382 3.383a.476.476 0 0 0-.149.328c0 .273.18.453.453.453a.47.47 0 0 0 .344-.149L10.25 9.82l.984-1.047-.078 1.282v1.718c0 .29.18.47.453.47Z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
